.github/workflows/solidity-foundry-artifacts.yml

```diff
@@ -0,0 +1,479 @@
+name: Solidity Foundry Artifact Generation
+on:
+  workflow_dispatch:
+    inputs:
+      product:
+        type: choice
+        description: 'product for which to generate artifacts; should be the same as Foundry profile'
+        required: true
+        options:
+          - "automation"
+          - "ccip"
+          - "functions"
+          - "keystone"
+          - "l2ep"
+          - "liquiditymanager"
+          - "llo-feeds"
+          - "operatorforwarder"
+          - "shared"
+          - "transmission"
+          - "vrf"
+      base_ref:
+        description: 'commit or tag to be used as base reference, when looking for modified Solidity files'
+        required: true
+
+env:
+  FOUNDRY_PROFILE: ci
+  # Has to match the `make foundry` version.
+  FOUNDRY_VERSION: nightly-de33b6af53005037b463318d2628b5cfcaf39916
+
+jobs:
+  changes:
+    name: Detect changes
+    runs-on: ubuntu-latest
+    outputs:
+      changes: ${{ steps.changes.outputs.sol }}
+      product_changes: ${{ steps.changes-transform.outputs.product_changes }}
+      files: ${{ steps.changes-transform.outputs.product_files }}
+      changeset_changes: ${{ steps.changes-dorny.outputs.changeset }}
+      changeset_files: ${{ steps.changes-dorny.outputs.changeset_files }}
+    steps:
+      - name: Checkout the repo
+        uses: actions/checkout@9bb56186c3b09b4f86b1c65136769dd318469633 # v4.1.2
+      - name: Find modified contracts
+        uses: AurorNZ/paths-filter@3b1f3abc3371cca888d8eb03dfa70bc8a9867629 # v4.0.0
```

Comment:
Curious why we're using two different "path filter" actions, this and the dorny one?

Reply:
dorny is using `picomatch` JS library, which behaves in a funny way, when you use a slice of exclusions. 

This:
```
- 'contracts/src/v0.8/<<product>>/*.sol'
- '!contracts/src/v0.8/<<product>>/tests/*t.sol'
- - '!contracts/src/v0.8/<<product>>/mocks/*t.sol'
```
will result in adding all sol files for automation, excluding tests and mocks, because 2nd and 3rd element will add everything that doesn't match these filters :/

The other actions is a fork of dorny that uses another JS library for pattern matching that handles all these filters with AND instead of OR.

Reply:
I think you can just added this to make it an AND:
https://github.com/smartcontractkit/chainlink/blob/develop/.github/workflows/changeset.yml#L38